src/server.rs

```diff
@@ -475,11 +475,26 @@ pub trait Output {
     }
 
     fn notify(&self, message: &str) {
-        let output = serde_json::to_string(
-            &NotificationMessage::new(message.to_owned(), ())
-        ).unwrap();
+        let output = serde_json::to_string(&NotificationMessage::new(message.to_owned(), ()))
+            .unwrap();
         self.response(output);
     }
+
+    fn notify_begin(&self) {
+        self.notify("rustDocument/diagnosticsBegin");
```

Comment:
I was thinking about something like compiler to LSP adapter, how do you find that ?